CMakeLists.txt

```diff
@@ -0,0 +1,2085 @@
+    endif()
+    target_link_libraries(mixxx-lib PUBLIC Silk::Float)
+  endif()
+endif()
+
+# MAD MP3 Decoder
+find_package(MAD)
+find_package(ID3Tag)
+cmake_dependent_option(MAD "MAD MP3 Decoder" ON "MAD_FOUND;ID3Tag_FOUND" OFF)
+if(MAD)
+  if(NOT MAD_FOUND)
+    message(FATAL_ERROR "MAD support requires libmad and its development headers.")
+  endif()
+  if(NOT ID3Tag_FOUND)
+    message(FATAL_ERROR "ID3Tag support requires libid3tag and its development headers.")
+  endif()
+  target_sources(mixxx-lib PRIVATE src/sources/soundsourcemp3.cpp)
+  target_compile_definitions(mixxx-lib PUBLIC __MAD__)
+  target_link_libraries(mixxx-lib PUBLIC MAD::MAD ID3Tag::ID3Tag)
+endif()
+
+# Mac App Store
+if(APPLE)
+  option(MACAPPSTORE "Build for Mac App Store" OFF)
+  if(MACAPPSTORE)
+    target_compile_definitions(mixxx-lib PUBLIC __MACAPPSTORE__)
+  endif()
+endif()
+
+# Media Foundation AAC Decoder Plugin
+find_package(MediaFoundation)
+cmake_dependent_option(MEDIAFOUNDATION "Media Foundation AAC decoder plugin (Windows Vista with KB2117917 or Windows 7 required)" ON "MediaFoundation_FOUND" OFF)
+if(MEDIAFOUNDATION)
+  if(NOT MediaFoundation_FOUND)
+      message(FATAL_ERROR "MediaFoundation support requires the MediaFoundation libraries and its development headers.")
+  endif()
+  target_sources(mixxx-lib PRIVATE
+    src/sources/soundsourcemediafoundation.cpp
+  )
+  target_compile_definitions(mixxx-lib PUBLIC __MEDIAFOUNDATION__)
+  target_include_directories(mixxx-lib SYSTEM PRIVATE
+    ${MediaFoundation_INCLUDE_DIRS}
+  )
+  target_link_libraries(mixxx-lib PUBLIC
+    ${MediaFoundation_LIBRARIES}
+  )
+endif()
+
+# Modplug support
+find_package(Modplug)
+cmake_dependent_option(MODPLUG "Modplug module decoder support" ON "Modplug_FOUND" OFF)
+if(MODPLUG)
+  if(NOT Modplug_FOUND)
+    message(FATAL_ERROR "Modplug module decoder support requires libmodplug and its development headers.")
+  endif()
+  target_sources(mixxx-lib PRIVATE
+    src/preferences/dialog/dlgprefmodplugdlg.ui
+    src/sources/soundsourcemodplug.cpp
+    src/preferences/dialog/dlgprefmodplug.cpp
+  )
+  target_compile_definitions(mixxx-lib PUBLIC __MODPLUG__)
+  target_link_libraries(mixxx-lib PUBLIC Modplug::Modplug)
+endif()
+
+# Profiling
+if(UNIX AND NOT APPLE)
+  option(PROFILING "Profiling (e.g. gprof) support" OFF)
+  if(PROFILING)
+    target_compile_options(mixxx-lib PUBLIC -pg)
+    target_link_options(mixxx-lib PUBLIC -pg)
+  endif()
+endif()
+
+# QtKeychain
+find_package(Qt5 QUIET OPTIONAL_COMPONENTS Keychain)
+option(QTKEYCHAIN "Secure credentials storage support for Live Broadcasting profiles" OFF)
+if(QTKEYCHAIN)
+  if(NOT Qt5_Keychain_FOUND)
+    message(FATAL_ERROR "Secure credential storage support requires the Qt5::Keychain component.")
+  endif()
+  target_compile_definitions(mixxx-lib PUBLIC __QTKEYCHAIN__)
+  target_link_libraries(mixxx-lib PUBLIC Qt5::Keychain)
+endif()
```

Comment:
This fails on OpenBSD:

```
$ cmake -DQt5_DIR=/usr/local/lib/qt5/cmake/Qt5/ -DQTKEYCHAIN=ON ..
...
CMake Error at CMakeLists.txt:1576 (message):
  Secure credential storage support requires the Qt5::Keychain component.
```

I think this is because `QtKeychain` is not part of `QT` and `Qt5KeychainConfig.cmake` is 
not located in the `Qt5_DIR`. 

The following diff makes it work for me:                                          

```
-find_package(Qt5 QUIET OPTIONAL_COMPONENTS Keychain)
+find_package(Qt5Keychain QUIET)
 option(QTKEYCHAIN "Secure credentials storage support for Live Broadcasting profiles" OFF)
 if(QTKEYCHAIN)
-  if(NOT Qt5_Keychain_FOUND)
+  if(NOT Qt5Keychain_FOUND)
     message(FATAL_ERROR "Secure credential storage support requires the Qt5::Keychain component.")
   endif()
   target_compile_definitions(mixxx-lib PUBLIC __QTKEYCHAIN__)
-  target_link_libraries(mixxx-lib PUBLIC Qt5::Keychain)
+  target_link_libraries(mixxx-lib PUBLIC ${QTKEYCHAIN_LIBRARIES})
 endif()

```
(I did not test it on Linux)

Reply:
Thanks. If using the non-target syntax for `Qt5Keychain`, don't we also need `target_include_directories`?

Reply:
> Thanks. If using the non-target syntax for `Qt5Keychain`, don't we also need `target_include_directories`?

Yes, this sounds reasonable, I guess it only works for me without it by chance.